pkg/messaging/mqtt/pubsub.go

```diff
@@ -87,18 +88,19 @@ func (ps pubsub) Subscribe(id, topic string, handler messaging.MessageHandler) e
 			if err != nil {
 				return err
 			}
-			if len(ps.subscriptions) == 0 {
+			if len(ps.subscriptions[id].topics) == 0 {
 				client, err := newClient(ps.address, id, ps.timeout)
 				if err != nil {
 					return err
 				}
 				s = subscription{
 					client: client,
-					topics: []string{topic},
+					topics: []string{},
 				}
 			}
 		}
 		s.topics = append(s.topics, topic)
```

Comment:
This will not work for the same reasons I mentioned before. `s` **is a copy of ps.subscriptions[id]** and will not be affected by `ps.Unsubscribe`. You can test by inspecting the `ps.subscriptions` value - there will be two "topic.engine" topics for the client with `clientid1`.